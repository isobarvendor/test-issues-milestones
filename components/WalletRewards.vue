<template>
  <div class="wallet-rewards" >
    <SliderHeader>{{cmsData.Wallets.RewardsTitle}}</SliderHeader>
    <div class="wallet-desktop" v-if="$mq == 'xl'">
      <div class="wallet-swiper" v-for="(item, index) in toBeShown" :key="'reward'+index">
        <div class="wallet-swiper-item">
          <img src="/img/prize-default.svg"
          />
        </div>
        <div class="description bg">
            <div class="name">{{item.currency_name}}</div>
            <div class="expiry">
              <span>365</span>
              <span>days left</span>
            </div>
          </div>
      </div>
    </div>
    <swiper class="swiper wallet-swiper" :options="swiperOption"  v-if="$mq == 'sm' || $mq == 'md' || $mq == 'lg'">
      <swiper-slide
        v-for="(item, index) in data.expirationWallet"
        :key="'reward'+index"
        class="wallet-swiper-item"
      >
        <img
          :src="item.imageMobile ? item.imageMobile.url : '/img/prize-default.svg'"
        />
        <div class="description bg">
          <div class="name">{{item.currency_name}}</div>
          <div class="expiry">
            <span>{{dayDiff(dates)}}</span>
            <span>days left</span>
          </div>
        </div>
      </swiper-slide>
      <div class="prize-swiper-pagination" slot="pagination"></div>
    </swiper>
    <a v-if="$mq == 'xl'" class="button center" @click="nextPage" :disabled="currentPage == totalPages">View all</a>
  </div>
</template>

<script>
import SliderHeader from "@/components/WalletSliderHeader";
import { Swiper, SwiperSlide } from "vue-awesome-swiper";
export default {
  name: "WalletRewards",
  components: {
    SliderHeader,
    Swiper,
    SwiperSlide,
  },
  data() {
    return {
      swiperOption: {
        slidesPerView: 2,
        spaceBetween: 10,
        breakpoints: {
          // when window width is >= 320px

          768: {
            slidesPerView: 3,
            spaceBetween: 20,
          },
          992: {
            slidesPerView: 3,
            spaceBetween: 30,
          },
        },
        freeMode: true,
        pagination: {
          el: ".prize-swiper-pagination",
          clickable: true,
          bulletClass: "prize-swiper-pagination-bullet",
          bulletActiveClass: "prize-swiper-pagination-bullet-active",
        }
      },
      currentPage: 1
    };
  },
  props: {
    data: null,
    cmsData:null
  },
  computed:{
  	toBeShown() {
    	return this.data.expirationWallet.slice(0, this.currentPage * 3);
    },
    totalPages() {
    	return Math.ceil( this.data.expirationWallet.length / 3);
    }
  },
  methods:{
    nextPage(){
    	if(this.currentPage <  this.totalPages) this.currentPage++;
    },
    prevPage(){
    	this.currentPage = this.currentPage - 1 || 1;
    },
    dayDiff(dates){
      let today = new Date();
      let dates2 = new Date(dates);
      let dateDiff = today.getDate() - dates2.getDate();
      return dateDiff
    }
  },
  beforeMount() {},
  mounted(){

  }
};
</script>

<style lang="scss">
.wallet-rewards{
    margin-bottom: 40px;
    .wallet-desktop{
      display: flex;
      flex-wrap: wrap;
      .wallet-swiper{
        margin: 30px 20px;
        width: 29%;
      }
      .description{
        padding: 20px;
        border-radius: 0px 0px 20px 20px;

        .name:after{
          content: '';
          display: block;
          margin: .5em 0;
          height: 1px;
          background: #585858;

        }
        .expiry{
          display: flex;
          justify-content: flex-end;
          span{
            font-size:14px;
          }
          span:first-child{
            margin-right: 0.5em;
            font-weight: bold;
            font-size: 26px;
          }
        }
      }
    }
    .prize-swiper-pagination > * + *:before{
      background: #585858
    }
    .prize-swiper-pagination .prize-swiper-pagination-bullet{
      background: #585858
    }
    .prize-swiper-pagination .prize-swiper-pagination-bullet-active{
      background: #ffffff
    }
    @media only screen and (max-width: 1199px) {
      .swiper.wallet-swiper{
        .description{
          padding: 1em 10px;
          .name{
            font-size: 14px;
          }
          .expiry{
            justify-content: flex-start;
            span:first-child{
              font-size: 14px;
            }
          }
        }

      }
    }

}
</style>
