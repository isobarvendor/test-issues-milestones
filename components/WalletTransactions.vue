<template>
  <div class="wallet-coins">
    <SliderHeader>Coin Transactions</SliderHeader>
    <div class="wallet-desktop">
      <div class="wallet-swiper" v-for="(item, index) in toBeShown" :key="'reward'+index">
        <div class="description bg">
            <div class="name">05 Nov 2019</div>
            <div class="collected">
              <span>Manchester United</span>
              <span class="negative">-50 points</span>
            </div>
            <div class="collected">
              <span>Coins Collected</span>
              <span class="positive">+10 points</span>
            </div>
          </div>
      </div>
    </div>
    <a class="button center" @click="nextPage" :disabled="currentPage == totalPages">View all</a>
  </div>
</template>

<script>
import SliderHeader from "@/components/WalletSliderHeader";
export default {
  name: "WalletTransactions",
  components: {
    SliderHeader
  },
  data() {
    return {
      currentPage: 1
    };
  },
  props: {
    data: null,
  },
  computed:{
  	toBeShown() {
    	return this.data.exclusivePrizes.slice(0, this.currentPage * 3);
    },
    totalPages() {
    	return Math.ceil( this.data.exclusivePrizes.length / 3);
    }
  },
  methods:{
    nextPage(){
    	if(this.currentPage <  this.totalPages) this.currentPage++;
    },
    prevPage(){
    	this.currentPage = this.currentPage - 1 || 1;
    }
  },
  beforeMount() {},
};
</script>
<style lang="scss">
  .wallet-coins{
    margin-bottom: 40px;
    .wallet-desktop{
      display: flex;
      flex-wrap: wrap;
      .wallet-swiper{
        margin: 30px 20px;
        width: 29%;
        .bg{
          background: linear-gradient(0deg, black 0%, #282828 100%);
          min-height: 140px;
        }
      }
      .description{
        padding: 20px;
        border-radius: 20px 20px 20px 20px;
        font-size: 14px;
        .name{
          font-size: 14px;
          color: #585858;
        }
        .name:after{
          content: '';
          display: block;
          margin: .5em 0;
          height: 1px;
          background: #585858;

        }
        .collected{
          display: flex;
          justify-content: space-between;
          margin-top: 10px;
          .negative{
            color: #DB1F27;
          }
          .positive{
            color: #FFC529;
          }
        }
      }
    }
    @media only screen and (max-width: 1199px) {
      .wallet-desktop{
        .wallet-swiper{
          width: 100%; 
          .bg{
            background: none;
            min-height: 140px;
            padding: 10px;
          }   
        }
      }  
    }
  }
</style>