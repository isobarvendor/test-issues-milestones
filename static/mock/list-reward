[ {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   }
]
