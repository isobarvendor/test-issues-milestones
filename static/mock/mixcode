{
                      "programId": 453269,
                      "code": "3WTD90GE5",
                      "lot": {
                        "lotId": 4455843436,
                        "lotName": "Coke 500ml PET",
                        "activateDate": 1611208800000,
                        "inactivateDate": 1627603200000,
                        "active": true,
                        "expired": false,
                        "bevProdPkg": {
                          "trademarkCd": "05",
                          "trademarkName": "Coca-Cola",
                          "brandCd": "001",
                          "brandName": "Coca-Cola",
                          "bevProdCd": "0001",
                          "bevProdName": "Coca-Cola",
                          "caffeinated": true,
                          "carbonated": true,
                          "calorieCategory": {
                            "code": 1,
                            "description": "Regular"
                          }
                        },
                        "releaseId": 494494,
                        "releaseName": "Coke PET",
                        "organizationId": 157168,
                        "organizationName": "Harish",
                        "realtimeCode": false,
                        "testRelease": false,
                        "pointValue": 0,
                        "bevProdOptions": []
                      },
                      "redeemed": false,
                      "actualCode": "3WTD90GE5"
                    }
