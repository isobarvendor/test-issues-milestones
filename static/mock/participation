{"id":16,"questionId":1,"answerId":3,"participationId":17}
