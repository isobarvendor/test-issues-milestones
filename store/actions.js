
import GeneralAPI from '../api/general';

import {LOGIN, SIGNUP, GET_ACCOUNT} from './action_types';

export default {
  [LOGIN]: ({ commit, state, getters }, data) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.login(data)
        .then(response => {
          commit('SET_TOKEN', response.data.token);
          sessionStorage.token=response.data.token;
          return resolve(messages);
        })
        .catch(error => {

          return reject(error);
        });
      }
    });
  },
  [GET_ACCOUNT]: ({ commit, state, getters }, token) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.getAccount(token)
        .then(response => {

          commit('SET_LOGIN_ACCOUNT', response.data);
          return resolve(messages);
        })
        .catch(error => {

          return reject(error);
        });
      }
    });
  },
  [SIGNUP]: ({ commit, state, getters }, data) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.signup(data)
        .then(messages => {
          //commit('SET_LOGIN_ACCOUNT', messages);
          return resolve(messages);
        })
        .catch(error => {
          console.error(error);
          return reject(error);
        });
      }
    });
  },


  /*[SET_GLOBAL_ERROR_DIALOG]:({commit, state}, status)=>{
    commit('SET_GLOBAL_ERROR_DIALOG', status);
  }
  */
};
