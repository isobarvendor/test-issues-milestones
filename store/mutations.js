

//shared store are concatenated to the central store
//TO-DO: manage the shared store seperately by module
export default {

  SET_LOGIN_ACCOUNT:(state, data)=>{
        state.login = data;
    },

  SET_TOKEN:(state, data)=>{
      state.token = data;
  },
  SET_ERROR:(state, data)=>{
    state.errorLogin = data;
},
  SET_LIST_PRIZE:(state, data)=>{
    state.listPrize = data;
  },
  SET_LIST_WALLET:(state, data)=>{
    state.listWallet = data;
  },
  SET_FILE_AMAZON:(state, data)=>{
    state.fileUploaded = data;
  },
  SET_CONFIG:(state,data)=>{
    state.config = data
  },
  SET_CMS_CONTENT:(state,data)=>{
    state.CMSContent = data
  }



};
