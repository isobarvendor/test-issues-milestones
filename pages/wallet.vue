<template>
  <div v-if="dataStatus.status == 200" id="main" class="wallet">
    <div class="wrapper">
      <div class="container">
        <div class="header">Your Wallet</div>
        <WalletRewards :data="listWalletExpired"/>
        <WalletTransactions :data="listWalletTransaction"/>
      </div>

    </div>
  </div>
  <div v-else-if="dataStatus.status >= 500">
    Status: {{dataStatus.status}}
    <br />
    {{dataStatus.message}}
  </div>
</template>

<script>
import WalletRewards from '@/components/WalletRewards'
import WalletTransactions from '@/components/WalletTransactions'
import { GET_LIST_WALLET } from '@/store/action_types';
export default {
  data(){
    return{
      dataStatus:{},
      data:null,
      configData: null,
      campaignType: 0,
      listWallet:[],
      listWalletExpired:[]
    }
  },
  head() {
    return {
      title: "Coke Campaign Title",
      meta: [
        {
          hid: "description",
          name: "description",
          content: "Home page description"
        }
      ],
      noscript: [],
      /*script: [
        {
          hid: "anime",
          src: "/js/anime.min.js"
        }
      ],*/
      css: []
    };
  },
  created() {
    this.fetchData();
    this.getListWallet();
    this.getListWalletExpired();
    this.getListWalletTransaction();
  },
  methods: {
    async fetchData() {
      let result = await this.$axios.get(
        "https://ayo.aircovery.com/cms-api/campaigns"
      );
      this.dataStatus = { status: result.status, message: result.statusText };
      this.data = result.data;

      let config = await this.$axios.get('https://ayo.aircovery.com/cms-api/campaign-configurations')
      this.configData = config.data
    },
     async getListWallet(){
     /* let request= {
              "configurationId": "confIdExample123",
              "flowLabel": "queryWallet",
              "userId": this.$store.state.login ? this.$store.state.login.uuid : null
          }*/
      let request = {
        name:this.$store.state.login.name,
        email:this.$store.state.login.email
      }
      //commented first after API ready
      let payload={request,token:this.$store.state.token}
      this.$store.dispatch(GET_LIST_WALLET,payload)
            .then((response)=>{

               this.listWallet=response.data;
            })
            .catch((error) =>{
              if(error.response && error.response.data.status=='401'){
                this.errorMessage='Please enter the correct email/password';
              }
            })

      /* this.listWallet={
                "walletStatus": [
                      {
                          "currencyId": "fl2gr1joh3nces",
                          "gppUserId": "UUID|CDS region",
                          "amount": 11,
                          "lastModified": 1542884788416

                      }, {
                          "currencyId": "79q3u1oysb",
                          "gppUserId": "UUID|CDS region",
                          "amount": 31,
                          "lastModified": 1542884788416

                      }

                  ]

              }*/

     },
        async getListWalletExpired(){
      let request= {
              "configurationId": "confIdExample123",
              "flowLabel": "queryExpirationWalletByUser",
             "userId": this.$store.state.login ? this.$store.state.login.uuid : null
          }
      //commented first after API ready

      let payload={request,token:this.$store.state.token}
      this.$store.dispatch(GET_LIST_WALLET,payload)
            .then((response)=>{

               this.listWalletExpired=response.data;
            })
            .catch((error) =>{
              if(error.response && error.response.data.status=='401'){
                this.errorMessage='Please enter the correct email/password';
              }
            })

       this.listWalletExpired={
                "expirationWallet": [
                    {
                        "gpp_user_id": "UUID|CDS region",
                        "amount": 20,
                        "currency_id": "coin",
                        "currency_name": "CCA-coin",

                        "configuration_id": "exampleConfig123",
                        "last_modified": 1557764018189,
                        "valid_thru": 1355228281846,
                        "already_spent": 15,
                        "spent_amount": 0
                    },
                    {
                        "gpp_user_id": "UUID|CDS region",
                        "amount": 1,
                        "configuration_id": "exampleConfig123",
                        "currency_id": "coin",
                        "currency_name": "CCA-coin",
                        "valid_thru": 1581811199000,
                        "already_spent": 0,
                        "spent_amount": 0
                    },
                    {
                        "gpp_user_id": "UUID|CDS region",
                        "amount": 40,
                        "currency_id": "coin",
                        "currency_name": "CCA-coin",

                        "configuration_id": "exampleConfig123",
                        "last_modified": 1557764018189,
                        "valid_thru": 1345228281846,
                        "already_spent": 0,
                        "spent_amount": 0
                    }
                ]
            }

     },
     async getListWalletTransaction(){
          let request=    {
                "configurationId": "1ejw41joh4lm23",
                "flowLabel": "addTransaction",
                "userId": this.$store.state.login ? this.$store.state.login.uuid : null,
                "eventCode": "testEventCode",
                "currencyAllocations":
                    [
                        {
                            "currencyId": "fl2gr1joh3nces",
                            "amount": 30
                        },
                        {
                            "currencyId": "79q3u1oysb",
                            "amount": 0
                        }
                    ]
            }

          //commented first after API ready

                 let payload={request,token:this.$store.state.token}
                 this.$store.dispatch(GET_LIST_WALLET,payload)
                      .then((response)=>{

                        this.listWalletTransaction=response.data;
                      })
                      .catch((error) =>{
                        if(error.response && error.response.data.status=='401'){
                          this.errorMessage='Please enter the correct email/password';
                        }
                      })

          this.listWalletTransaction ={
              "transactions": [
                  {
                      "transactionInserted": true,
                      "currencyId": "fl2gr1joh3nces",
                      "walletRollingTotal": 30
                  },
                  {
                      "transactionInserted": true,
                      "currencyId": "79q3u1oysb",
                      "walletRollingTotal": 0
                  }
              ]
          }

     }
  },
};
</script>

<style lang="scss">
@import '~assets/scss/variables.scss';
  html, body{
    background: $body-bg-color-wallet;
    @media only screen and (max-width: 767px) {
      .wallet{
        .header{
          font-size: 25px;
          margin-bottom: 30px;
        }
      }
    }
  }
</style>
