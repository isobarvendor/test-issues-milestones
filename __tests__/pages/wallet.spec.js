import { mount,  createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router'

import wallet from '@/pages/wallet';

import WalletRewards from '@/components/WalletRewards'
import WalletTransactions from '@/components/WalletTransactions'


import {mockCMSConfig,mockCMSContent} from "@/__mock__/mock";


import actions from '@/store/actions';
import mutations from '@/store/mutations';


let localVue, store,router;


  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(VueRouter);
    router = new VueRouter();
    store = new Vuex.Store({
      getters: {
        getCMSContent() {
          return mockCMSContent;
        },
        getLoginAccount(){
          return {
            name:"ron",
            email:"ron@dsw.com",
            imageUrl:"tes"
          }
        },
        getCMSConfig(){
          return mockCMSConfig[0];
        }
      },
      state:{
        token:"",
        language: "en",
        listWallet:{
          walletStatus:[
            {amount :30}
          ]
        },

      },
      actions,
      mutations
    });
  });

  const factory = () => {
    return mount(wallet, {

      localVue, store,router
    });
  };



describe('wallet', () => {


  test('renders a WalletRewards', () => {
    const wrapper = factory();
    const component= wrapper.findComponent(WalletRewards);
    expect(component.exists()).toBeTruthy();
  });
  test('renders a WalletTransactions', () => {
    const wrapper = factory();
    const component= wrapper.findComponent(WalletTransactions);
    expect(component.exists()).toBeTruthy();
  });


  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

});
