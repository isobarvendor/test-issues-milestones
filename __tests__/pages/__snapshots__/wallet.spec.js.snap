// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wallet renders properly 1`] = `
<div id="main" class="wallet">
  <div class="wrapper">
    <div class="container">
      <div class="header">My wallet</div>
      <div class="wallet-rewards">
        <div class="slider-header">
          <div class="title">Rewards</div>
        </div>
        <!---->
        <!---->
        <!---->
      </div>
      <div class="wallet-coins">
        <div class="slider-header">
          <div class="title">Coin Transactions</div>
        </div>
        <div class="wallet-desktop">
          <div class="wallet-swiper">
            <div class="description bg">
              <div class="name">05 Nov 2019</div>
              <div class="collected"><span>Manchester United</span> <span class="negative">30 points</span></div>
            </div>
          </div>
          <div class="wallet-swiper">
            <div class="description bg">
              <div class="name">05 Nov 2019</div>
              <div class="collected"><span>Manchester United</span> <span class="negative">0 points</span></div>
            </div>
          </div>
        </div> <a disabled="disabled" class="button center">View all</a>
      </div>
    </div>
  </div>
</div>
`;
